src/ConnectedMode.UnitTests/ServerSentEvents/SSESessionManagerTests.cs

```diff
@@ -0,0 +1,230 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Threading.Tasks;
+using SonarLint.VisualStudio.ConnectedMode.ServerSentEvents;
+using SonarLint.VisualStudio.Core.Binding;
+using SonarLint.VisualStudio.Integration.UnitTests;
+
+namespace SonarLint.VisualStudio.ConnectedMode.UnitTests.ServerSentEvents;
+
+[TestClass]
+public class SSESessionManagerTests
+{
+    [TestMethod]
+    public void MefCtor_CheckIsExported()
+    {
+        MefTestHelpers.CheckTypeCanBeImported<SSESessionManager, ISSESessionManager>(
+            MefTestHelpers.CreateExport<IActiveSolutionBoundTracker>(),
+            MefTestHelpers.CreateExport<ISSESessionFactory>());
+    }
+
+    [TestMethod]
+    public void Ctor_SubscribesToSolutionEvents()
+    {
+        var solutionTrackerMock = new Mock<IActiveSolutionBoundTracker>();
+
+        var testSubject = new SSESessionManager(solutionTrackerMock.Object, Mock.Of<ISSESessionFactory>());
+
+        solutionTrackerMock.VerifyAdd(
+            solutionTracker => solutionTracker.SolutionBindingChanged +=
+                It.IsAny<EventHandler<ActiveSolutionBindingEventArgs>>(), Times.Once);
+    }
+
+    [DataTestMethod]
+    [DataRow(true)]
+    [DataRow(false)]
+    public void OnSolutionChanged_CorrectlyIdentifiesOpenAndClose(bool isOpen)
```

Comment:
But the copy paste is going to end up in almost every test 😞 Can you elaborate on why it's hard to understand? Maybe we can find a compromise

Reply:
Couple of things:
* OpenNewProject and CloseProject are not the right names for what's happening. The event is bindingChanged and it's either Standalone or Connected, there is no concept of project (nor a solution). The session manager only knows that the binding changed and so the tests need to speak in the same terminology.
* Names such as SetUpMocksDispose and SetUpMockConnection -- it's not clear what they're doing, especially when they work with sequences. 
* `sessionFactory.Create(It.IsAny<string>())` it is always better to set up with specific arguments. It.IsAny is used here because this is a generic function that receives the sequence, but it makes it more complicated for the reader to understand, versus is seeing a specific set up in the code and then immediately the verification. 

I think it's ok that there is some duplication in the unit tests because it increases the readability of every test. When there are so few functions that encapsulate everything, it is very difficult to understand what actually is being tested.

Reply:
IMO points 1 and 2 are just bad naming and point 3 can be fixed as well. "When there are so few functions that encapsulate everything" - but if you get rid of the class, you either keep the same functions but move them to the test class or get rid of them and get a lot of duplication. "it is very difficult to understand what actually is being tested." - it's in the name of the test method + verifications are exactly the same except the `testScope.*` prefix

Reply:
Ok, try to rename and let's see. IMO the code I sent you was more explicit and easier to understand than the original.